### PR TITLE
[16.0] [MIG] maintenance_plan

### PR DESCRIPTION
**Pull request by alexeirivera87**

### CODE REVIEW

File: maintenance_plan/tests/test_maintenance_plan.py [hunk 1]
```diff
@@ -228,10 +228,10 @@ def test_maintenance_request_report(self):
...
         )
         generated_request.note = "TEST-INSTRUCTIONS"
-        res = self.report_obj._get_report_from_name(
-            "base_maintenance.report_maintenance_request"
-        )._render_qweb_text(generated_request.ids, False)
-        self.assertRegex(str(res[0]), "TEST-INSTRUCTIONS")
+        rendering, qweb_type = self.report_obj._render_qweb_text(
+            "base_maintenance.report_maintenance_request", generated_request.ids
+        )
+        self.assertFalse(rendering.decode("utf-8").find("TEST-INSTRUCTIONS") == -1)
```
Comment (etobella) [inline maintenance_plan/tests/test_maintenance_plan.py]: REGEX was not working?

File: maintenance_plan/views/maintenance_equipment_views.xml [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+                        <field name="duration" />
+                        <field name="next_maintenance_date" />
+                        <field
+                            name="maintenance_plan_horizon"
+                            string="P. Horizon period"
+                        />
+                        <field name="planning_step" string="P. Horizon step" />
+                    </tree>
+                </field>
+            </xpath>
```
Comment (HekkiMelody) [inline maintenance_plan/views/maintenance_equipment_views.xml:48]: Not sure why the change from 

`<xpath expr="//group[@name='maintenance']/" position="after">`

`<xpath expr="//group[@name='maintenance']/.." position="after">`

at the very least I would add a `<group string="Maintenance Plans"> ` around the whole thing, or alternatively remove the `nolabel` option

**Review by HekkiMelody — COMMENTED** [review]
Overall looks good and seems to work, unsure about the reason for the few changes that were made compared to v15

**Comment by HekkiMelody** [comment]
Also there was an existing proposed migration #324 , make sure to check if someone has done the work already, or explicitly point out that you're superseeding the previous PR, please!

**Comment by mamcode** [comment]
> Also there was an existing proposed migration #324 , make sure to check if someone has done the work already, or explicitly point out that you're superseeding the previous PR, please!

There is also another duplicated PR that proposes to migrate this module to v16. #265

**Comment by mcassuto** [comment]
@aleuffre @alexeirivera87 @mamcode 
Hi guys since there is 3 PR in total, what should we do now ?

**Comment by HekkiMelody** [comment]
Leave a review on all of them, or the one you think is closest to being merged, both in terms of code and in terms of the author being more active.

In terms of code I think this one is closest to being done, there are just a few changes to be made.

Happy to open a 4th PR myself, since I'd like to have this merged ASAP :rofl:

**Comment by mcassuto** [comment]
Ok let me check if it has been tested ok, I guess it is alreay done on production.

**Comment by alexeirivera87** [comment]
Hello @mcassuto  @aleuffre @mamcode 

Yes, I did check the existing PR before submitting this one:

About #324 , on this migration the only thing the author did was to change the module version (and even this is not correct), and to remove a function that was failing in the tests.

About #265 , here the only thing that was done was to change the version of the module.

Regards

**Comment by HekkiMelody** [comment]
> Hello @mcassuto @aleuffre @mamcode
> 
> Yes, I did check the existing PR before submitting this one:
> 
> About #324 , on this migration the only thing the author did was to change the module version (and even this is not correct), and to remove a function that was failing in the tests.
> 
> About #265 , here the only thing that was done was to change the version of the module.
> 
> Regards

In that case, we usually write `superseds #number of the MR` in the description, so we can keep track of such things. Thank you :pray:

**Comment by etobella** [comment]
Visually it works for me. Why did  you remove the Regex on the test?

**Comment by HekkiMelody** [comment]
@alexeirivera87 if you could please address the changes that were requested, they're minor things and I think this is pretty close to getting merged.

if you don't have time, I'd like to pick it up myself (I'd preserve ownership of your commits) since I need this module merged as well.

Thank you for your contributions!

**Comment by etobella** [comment]
@alexeirivera87 can you check the issue with the REGEX?

**Comment by alexeirivera87** [comment]
Hello @aleuffre @etobella 

I reverted the changes in unit tests, so assertRegex is used as before.

Now I have to check why the test failed in method test_next_maintenance_date_02

Regards

**Comment by etobella** [comment]
But the test seems unrelated to the regex. Can you check it¿

**Comment by alexeirivera87** [comment]
@etobella All tests passed.

Yes, the is related to dates. Some are failing when current date is January 30th or 31st.

regards

**Comment by etobella** [comment]
Wow!!! I see... that might happend every 4 years :laughing: 

One option would be to use freezetime in order to avoid this. Can you try?

**Comment by victoralmau** [comment]
Hi, sorry for being late.
Added freeze_time in the tests in v15 https://github.com/OCA/maintenance/pull/371 but I see there are v15 changes missing in the commit history, can you add it?

The problem with the last days of the month is that the last day of the following month is not used (in those that are generated).

**Comment by alexeirivera87** [comment]
@victoralmau  @etobella 

It should be ok now.

regards

**Review by etobella — APPROVED** [review]

**Review by victoralmau — APPROVED** [review]
Code review OK
